app/post.go

```diff
@@ -586,6 +596,45 @@ func (a *App) GetPostsAroundPost(postId, channelId string, offset, limit int, be
 	}
 }
 
+func (a *App) GetPostsForChannelAroundLastUnread(channelId, userId string) (*model.PostList, *model.AppError) {
+	var member *model.ChannelMember
+	var err *model.AppError
+	if member, err = a.GetChannelMember(channelId, userId); err != nil {
+		return nil, err
+	}
+
+	var postListSince *model.PostList
+	if member.LastViewedAt > 0 {
+		if postListSince, err = a.GetPostsSinceWithLimit(channelId, member.LastViewedAt, 60); err != nil {
+			return nil, err
+		}
+	}
+
+	var lastUnreadPostId string
+	if len(postListSince.Order) > 0 {
```

Comment:
Similarly, I think we could exist early if `postListSince` is empty since I think that means none of the later code actually has any effect.

I think this one might need a test as well.

Reply:
👍.  Test covered it already although not explicitly set.  Anyway, I've modified the test.